src/shared/install.c

```diff
@@ -1338,7 +1338,10 @@ static int unit_file_load_or_readlink(
         if (r < 0)
                 return r;
 
-        if (path_equal(target, "/dev/null"))
+        if (!path_is_absolute(target))
+                target_fqpn = file_in_same_dir(path, target);
```

Comment:
file_in_same_dir() says:
>  This removes the last component of path and appends filename

which is not what is needed here. We need to resolve `target` relative to `root_dir`.
```C
        r = chase_symlinks_and_stat(path, root_dir, 0, NULL, &st, NULL);
        if (null_or_empty(st))
               ...
```
(`CHASE_PREFIX_ROOT` might be useful depending on how `unit_file_load` returns the result. Also possibly `CHASE_WARN` to increase logging.)

Reply:
Ok I get what `root_dir` is for now. Thanks.

I've made the update as you suggested but I'm wondering if this is going to break stuff? In the test I had to add `<root_dir>/dev/null` to appease an existing test that was expecting to only have to test against the string "/dev/null" and not the actual contents.

Reply:
I think @poettering suggested (or made patch somewhere?) to check for /dev/null literally before quering the fs. I think we should do this, at least in the case where we're operating offline.

Reply:
I pushed a patch (to your branch) to allow /dev/null to non exist under root. PTAL.

Reply:
Thanks @keszybz  the updated changes LGTM!